sys/linux/rdma.txt

```diff
@@ -0,0 +1,1464 @@
+define IB_USER_VERBS_EX_CMD_DESTROY_FLOW	0x80000033
+define MLX5_CREATE_DCT	0x8
+define MLX5_CREATE_DCI	0x16
+define IB_USER_VERBS_CMD_FLAG_EXTENDED	0x80000000
+define IB_USER_VERBS_CMD_FLAGS_SHIFT	24
+define IB_USER_VERBS_CMD_THRESHOLD	50
+define IB_USER_VERBS_CMD_CREATE_CQ	0x12
+define IB_USER_VERBS_CMD_CREATE_QP	0x18
+define EX_CREATE_CQ_CMD	IB_USER_VERBS_CMD_FLAG_EXTENDED | IB_USER_VERBS_CMD_CREATE_CQ
+define EX_CREATE_QP_CMD	IB_USER_VERBS_CMD_FLAG_EXTENDED | IB_USER_VERBS_CMD_CREATE_QP
+
+# flags
+rdma_dev_open_flags = O_RDWR, O_CLOEXEC
+ib_access_flags = IB_ACCESS_LOCAL_WRITE, IB_ACCESS_REMOTE_WRITE, IB_ACCESS_REMOTE_READ, IB_ACCESS_REMOTE_ATOMIC, IB_ACCESS_MW_BIND, IB_ZERO_BASED, IB_ACCESS_ON_DEMAND
+ib_mr_rereg_flags = IB_MR_REREG_TRANS, IB_MR_REREG_PD, IB_MR_REREG_ACCESS, IB_MR_REREG_SUPPORTED
+ib_qp_type = IB_QPT_SMI, IB_QPT_GSI, IB_QPT_RC, IB_QPT_UC, IB_QPT_UD, IB_QPT_RAW_IPV6, IB_QPT_RAW_ETHERTYPE, IB_QPT_RAW_PACKET, IB_QPT_XRC_INI, IB_QPT_XRC_TGT
+ib_qp_create_flags = IB_QP_CREATE_IPOIB_UD_LSO, IB_QP_CREATE_BLOCK_MULTICAST_LOOPBACK, IB_QP_CREATE_CROSS_CHANNEL, IB_QP_CREATE_MANAGED_SEND, IB_QP_CREATE_MANAGED_RECV, IB_QP_CREATE_NETIF_QP, IB_QP_CREATE_SIGNATURE_EN, IB_QP_CREATE_SCATTER_FCS
+ib_send_flags = IB_SEND_FENCE, IB_SEND_SIGNALED, IB_SEND_SOLICITED, IB_SEND_INLINE, IB_SEND_IP_CSUM
+ib_flow_flags = IB_FLOW_ATTR_FLAGS_DONT_TRAP
+ib_ipv4_flags = IB_IPV4_DONT_FRAG, IB_IPV4_MORE_FRAG
+mlx5_create_qp_flags = MLX5_QP_FLAG_SIGNATURE, MLX5_QP_FLAG_SCATTER_CQE, MLX5_QP_FLAG_TUNNEL_OFFLOADS, MLX5_QP_FLAG_BFREG_INDEX, MLX5_CREATE_DCT, MLX5_CREATE_DCI
+mlx5_create_cq_flags = 0
+create_cq_ex_flags = IB_UVERBS_CQ_FLAGS_TIMESTAMP_COMPLETION, IB_UVERBS_CQ_FLAGS_IGNORE_OVERRUN
+create_cq_ex_mask = 0x0, 0x1
+mlx5_comp_cqe_res_format = MLX5_IB_CQE_RES_FORMAT_HASH, MLX5_IB_CQE_RES_FORMAT_CSUM
+wq_type = IB_WQT_RQ
+create_wq_flags = IB_WQ_FLAGS_CVLAN_STRIPPING, IB_WQ_FLAGS_SCATTER_FCS, IB_WQ_FLAGS_DELAY_DROP, IB_WQ_FLAGS_PCI_WRITE_END_PADDING
+modify_wq_attr_flags = IB_WQ_STATE, IB_WQ_CUR_STATE, IB_WQ_FLAGS
+modify_wq_flags = IB_WQ_FLAGS_CVLAN_STRIPPING, IB_WQ_FLAGS_SCATTER_FCS, IB_WQ_FLAGS_DELAY_DROP, IB_WQ_FLAGS_PCI_WRITE_END_PADDING
+srq_type = IB_SRQT_BASIC, IB_SRQT_XRC, IB_SRQT_TM
+create_srq_ex_flags = MLX5_SRQ_FLAG_SIGNATURE
+
+# structs
+# =======
+
+# context
+# ===========
+
+mlx5_get_context_cmd_resp {
+	async_fd			int32
+	num_comp_vectors		int32
+	qp_tab_size			int32
+	bf_reg_size			int32
+	tot_uuars			int32
+	cache_line_size			int32
+	max_sq_desc_sz			int16
+	max_rq_desc_sz			int16
+	max_send_wqebb			int32
+	max_recv_wr			int32
+	max_srq_recv_wr			int32
+	num_ports			int16
+	reserved1			const[0x0, int16]
+	comp_mask			int32[0:1]
+	response_length			int32
+	cqe_version_cmds_supp_uhw	int16
+	reserved2			const[0x0, int16]
+	hca_core_clock_offset		int64
+	log_uar_size			int32
+	num_uars_per_page		int32
+	num_dyn_bfregs			int32
+	reserved3			const[0x0, int32]
+}
+
+mlx5_get_context_cmd {
+	command			const[0x0, int32]
```

Comment:
Are there any literal consts for these? Is it IB_USER_VERBS_CMD_GET_CONTEXT? Please use literal consts.
It's hard for me to find any traces of this in kernel source to do any checking or understand what happens.
Usually if I see, say, `mlx5_get_context_cmd` I can do https://elixir.bootlin.com/linux/v5.1-rc1/ident/mlx5_get_context_cmd and find corresponding kernel code. But in this case I find nothing.
While https://elixir.bootlin.com/linux/v5.1-rc1/ident/IB_USER_VERBS_CMD_GET_CONTEXT gives something.